scipy/sparse/_construct.py

```diff
@@ -115,9 +110,9 @@ def diags(diagonals, offsets=0, shape=None, format=None, dtype=None):
 
```

Comment:
We need to update versionadded to be 1.11 here (and keep it as 0.11 in the old version). Unfortunately that might mean we have to duplicate the docstring instead of using the `_array_doc_to_matrix` magic.

Reply:
Indeed - I was hoping we could get away with the find-replace, but there are too many subtle differences like this.